src/components/MCQ/MCQQuestion.vue

```diff
@@ -31,27 +31,43 @@ import { ref } from "vue";
 import type { MCQuestion, MCQOptions } from "@type/MCQ.d.ts";
 import MCQOption from "./MCQOption.vue";
 import MCQButton from "./MCQButton.vue";
+import { useQuizStore } from "@/store/QuizStore";
 
-const { statement, optionsList } = defineProps<MCQuestion>();
+const statUpdate = useQuizStore();
+
+const { statement, optionsList, _id } = defineProps<MCQuestion>();
 const selectedOption = ref<string | null>(null);
+const correct_index = ref<string | null>(null);
 const submitted = ref<boolean>(false);
 const emit = defineEmits(["nextQuestion", "skipQuestion"]);
 
 const submitAnswer = () => {
   submitted.value = true;
 };
 
-const nextQuestion = () => {
-  resetQuestion();
+const nextQuestion = (_id: { $oid: string }) => {
+  resetQuestion(_id);
   emit("nextQuestion");
 };
 
 const skipQuestion = () => {
-  resetQuestion();
+  resetQuestion(_id);
   emit("skipQuestion");
 };
 
-const resetQuestion = () => {
+//function to emit correct answer count
+const countCorrect = (_id: { $oid: string }) => {
```

Comment:
I would refactor this code to look like this: 

```typescript
const countCorrect = (_id: { $oid: string }) => {
  const isOptionCorrect = () => optionsList[parseInt(selectedOption.value as string)].optionCorrect;

  if (submitted.value){
    isOptionCorrect() && statUpdate.incrementStat(_id.$oid, "correct");
    statUpdate.incrementStat(_id.$oid, "attempts");
  } else {
    statUpdate.incrementStat(_id.$oid, "skipped");
  }
};
```

I believe if we submit, regardless, we should have an attempt counted against it.

Reply:
Where we are going to count if a correct option been chosen? This changes seems not track the option correctly, run test failed and run manually also...

Reply:
The state should do the counting if that question is correct - the component doesn't need to know rather than the state.

When you import, you have the correct answer on hand because you get given an optionList.

We need to make sure we are utilising what we already have rather than creating a new local state.